src/OpenTracing.AspNetCore/OpenTracingMiddleware.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+
+namespace OpenTracing.AspNetCore
+{
+    public class OpenTracingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly OpenTracingOptions _options;
+
+        public OpenTracingMiddleware(RequestDelegate next, OpenTracingOptions options)
+        {
+            if (next == null)
+            {
+                throw new ArgumentNullException(nameof(next));
+            }
+
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            _next = next;
+            _options = options;
+        }
+
+        public Task Invoke(HttpContext context, ITracer tracer)
+        {
+            if (tracer == null)
+            {
+                throw new ArgumentNullException(nameof(tracer));
+            }
+
+            // see if this request is already part of a trace
+            var existingSpanContext = tracer.ExtractHttpHeaders(context);
+            ISpan requestSpan = null;
+
+            try
+            {
+                // TODO peer tags?
+
+                requestSpan = tracer.StartSpan(context.Request.Path, SpanReference.ChildOf(existingSpanContext))
+                    .SetTagComponent("AspNetCore")
+                    .SetTagSpanKindServer()
```

Comment:
see comment above - AspNetCore is just a PoC and will be moved elsewhere (to be defined)